src/Computer.php

```diff
@@ -293,16 +294,43 @@ public function post_updateItem($history = 1)
     public function prepareInputForAdd($input)
     {
 
+        //without right, use can't set remote_addr
+        if (isset($input['remote_addr']) && !Session::haveRight(Conf::$rightname, Conf::IMPORTFROMFILE)) {
+            unset($input['remote_addr']);
+        }
+
         if (isset($input["id"]) && ($input["id"] > 0)) {
             $input["_oldID"] = $input["id"];
         }
         unset($input['id']);
         unset($input['withtemplate']);
 
+        return $this->manageInput($input);
+    }
+
+
+    public function prepareInputForUpdate($input)
+    {
+        return $this->manageInput($input);
+    }
+
+
+    public function manageInput($input)
+    {
+        //without right, user can't set remote_addr
+        if (isset($input['remote_addr']) && !Session::haveRight(Conf::$rightname, Conf::IMPORTFROMFILE)) {
```

Comment:
I'm not sure on the right check

Reply:
Also, the check is not the same in twig template

Reply:
We have two right on inventory 

- IMPORTFROMFILE (having the right to inject means having the right to modify the data in the inventory  (IMHO)
- UPDATECONFIG

Maybe we could check on UPDATECONFIG

Reply:
@orthagh @cedric-anne any idea ?

Reply:
ping?

Reply:
IMHO, we should check `UPDATECONFIG`, as in twig template.

Reply:
after thinking about it, 
we should have the same behavior as with the management ports :
no check on the rights, if the user wants to modify afterwards he can

Reply:
If we do not prevent user to update it if they have no rights on inventory, it means that they will be able to add an unexpected IP on which they may receive agent requests later. I do not know if it can be considered as a security flaw.

Reply:
Inventory rights should concern inventory only, not direct edition. In that case, we should ask ourselves the same question if the user change any linked port, or serial, or uuid, or whatever the inventory uses to determine computer updates.